### PR TITLE
Fix incorrect query and args handling

### LINKED ISSUE #10: Bug: query arguments are added as a tag even when `WithOmitArgs()` is set
#### Expected

When the `WithOmitArgs()` is set, I expect query arguments to not be included in any of the spans.

#### Current behaviour

The arg tag is correctly omitted from the `sql-conn-exec` span, but it is included in the `sql-stmt-exec`. This may be specifically related to named queries?

image

_Cc: @luna-duclos_

### PR DESCRIPTION
**Pull request by luna-duclos**
Closes #10

### CODE REVIEW

**Comment by aaslamin** [comment]
Still plumbing this in my project to test, but looking at the changes I don't think this is what you want to do?

We still want to create the `arg` tag in our spans if the `OmitArgs` is set to false. Logging is a separate thing all together. In order to this, we must conditionally call `span.SetLabel("args", formatArgs(args))`. Maybe I am missing something obvious? 😅

**Comment by aaslamin** [comment]
Eh...Luna, apologies for the delayed response. I have not been able to build my project with a version of your package that has this fix to try it out. My project is using the new `go modules` system, I am not quite familiar with it yet. Recently migrated away from dep which made it trivial to build against a fork.

**Comment by aaslamin** [comment]
**Update**: I was able to build my project against your branch after battling it out a bit more about modules.

I can confirm this is not the behaviour we want as the tag is removed even if `WithOmitArgs()` is not set. Instead we want to conditionally remove the tag like it was being done in some places before. Also the query tag is removed all together:

image

File: sql.go [hunk 1]
```diff
@@ -302,10 +298,6 @@ func (c wrappedConn) Query(query string, args []driver.Value) (driver.Rows, erro
 func (c wrappedConn) QueryContext(ctx context.Context, query string, args []driver.NamedValue) (rows driver.Rows, err error) {
 	span := c.GetSpan(ctx).NewChild(OpSQLConnQuery)
 	span.SetLabel("component", "database/sql")
-	span.SetLabel("query", query)
-	if !c.OmitArgs {
-		span.SetLabel("args", formatArgs(args))
-	}
```
Comment (aaslamin) [inline sql.go]: So...this was previously good! We don't want to remove this check as far as I understand. Instead we want to do this in other places which are adding the `args` tag such as `ExecContext` and `QueryContext`.

Comment (luna-duclos) [reply sql.go]: That's odd ... the call to `logQuery` a bit later is supposed to take care of that, I'll look into why that is broken

**Comment by luna-duclos** [comment]
Issues fixed, fix corrected, merging this :)

**Comment by aaslamin** [comment]
😍 Yay, thanks Luna! 🏆

**Pull request merged**